### ISSUE TITLE
e2e test are flaky

### ISSUE THREAD

**Issue by vprashar2929** [sustainable-computing-io/kepler-operator#179]
After merging #175 the integration test CI [run](https://github.com/sustainable-computing-io/kepler-operator/actions/runs/5976922805/job/16215800980#step:7:377) reports an error but re-running the same job again resolves it. This flakiness in the e2e test should not happen

**Comment by sthaha** [comment]
@vprashar2929 , i think it is not the test that is being flaky but rather it is pointing to the fragility of the reconciliation loop :)

**Comment by sthaha** [comment]
https://github.com/sustainable-computing-io/kepler-operator/actions/runs/6006439663/job/16290969036#step:8:505
Shows that the namespace is taking a while to be deleted. 
```
🔔 Operator Error Logs

2023-08-29T00:37:10Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:15Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:20Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:25Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:30Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:35Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:40Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:40Z: Get after Delete: /openshift-kepler-operator Error: <nil> 
2023-08-29T00:37:40Z	LEVEL(-3)	continue reconciliation despite error	{"controller": "kepler", "controllerGroup": "kepler.system.sustainable.computing.io", "controllerKind": "Kepler", "Kepler": {"name":"kepler"}, "namespace": "", "name": "kepler", "reconcileID": "49d65ea3-2d18-400d-9513-6071b009f91c", "error": "/openshift-kepler-operator (/v1, Kind=Namespace): deleter: timed out waiting for deletion : timed out waiting for the condition"}
2023-08-29T00:37:40Z: Get after Delete: /kepler-exporter-scc Error: SecurityContextConstraints.security.openshift.io "kepler-exporter-scc" not found 
2023-08-29T00:37:40Z: Get after Delete: /kepler-exporter-cr Error: ClusterRoleBinding.rbac.authorization.k8s.io "kepler-exporter-cr" not found 
2023-08-29T00:37:40Z: Get after Delete: /kepler-exporter-cr Error: ClusterRole.rbac.authorization.k8s.io "kepler-exporter-cr" not found 
2023-08-29T00:37:40Z	INFO	Warning: Reconciler returned both a non-zero result and a non-nil error. The result will always be ignored if the error is non-nil and the non-nil error causes reqeueuing with exponential backoff. For more details, see: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler	{"controller": "kepler", "controllerGroup": "kepler.system.sustainable.computing.io", "controllerKind": "Kepler", "Kepler": {"name":"kepler"}, "namespace": "", "name": "kepler", "reconcileID": "49d65ea3-2d18-400d-9[513](https://github.com/sustainable-computing-io/kepler-operator/actions/runs/6006439663/job/16290969036#step:8:514)-6071b009f91c"}
2023-08-29T00:37:40Z	ERROR	Reconciler error	{"controller": "kepler", "controllerGroup": "kepler.system.sustainable.computing.io", "controllerKind": "Kepler", "Kepler": {"name":"kepler"}, "namespace": "", "name": "kepler", "reconcileID": "49d65ea3-2d18-400d-9513-6071b009f91c", "error": "/openshift-kepler-operator (/v1, Kind=Namespace): deleter: timed out waiting for deletion : timed out waiting for the condition"}
2023-08-29T00:37:41Z	LEVEL(-3)	continue reconciliation despite error	{"controller": "kepler", "controllerGroup": "kepler.system.sustainable.computing.io", "controllerKind": "Kepler", "Kepler": {"name":"kepler"}, "namespace": "", "name": "kepler", "reconcileID": "eb9acd4c-2181-4847-a9d2-9cd7ba0be101", "error": "kepler-exporter-sa (/v1, Kind=ServiceAccount): updater: patch failed : serviceaccounts \"kepler-exporter-sa\" is forbidden: unable to create new content in namespace openshift-kepler-operator because it is being terminated"}
2023-08-29T00:37:41Z	LEVEL(-3)	continue reconciliation despite error	{"controller": "kepler", "controllerGroup": "kepler.system.sustainable.computing.io", "controllerKind": "Kepler", "Kepler": {"name":"kepler"}, "namespace": "", "name": "kepler", "reconcileID": "eb9acd4c-2181-4847-a9d2-9cd7ba0be101", "error": "kepler-exporter-cm (/v1, Kind=ConfigMap): updater: patch failed : configmaps \"kepler-exporter-cm\" is forbidden: unable to create new content in namespace openshift-kepler-operator because it is being terminated"}
2023-08-29T00:37:41Z	LEVEL(-3)	continue reconciliation despite error	{"controller": "kepler", "controllerGroup": "kepler.system.sustainable.computing.io", "controllerKind": "Kepler", "Kepler": {"name":"kepler"}, "namespace": "", "name": "kepler", "reconcileID": "eb9acd4c-2181-4847-a9d2-9cd7ba0be101", "error": "kepler-exporter-ds (apps/v1, Kind=DaemonSet): updater: patch failed : daemonsets.apps \"kepler-exporter-ds\" is forbidden: unable to create new content in namespace openshift-kepler-operator because it is being terminated"}
```

**Comment by sthaha** [comment]
Closing this as fixed in #182
